README.md

```diff
@@ -1,45 +1,601 @@
+- [Refund online payment](#refund-online-payment): `PaymentClient.refundOnlinePayment` - refund an online payment
 
-Add the following configuration to your `pom.xml` file:
+Follow the links for these methods to see the sample code and the data specification.
 
-```xml
-<dependency>
-    <groupId>com.kodypay.grpc</groupId>
-    <artifactId>kody-clientsdk-java</artifactId>
-    <version>0.0.3</version>
-</dependency>
+## API data reference and Demo code
+
+Every request to the client library requires authentication and the identifier of the store. See more [authentication](#authenticate-to-payments-api).
+
+### Create online payment
+
+Send a payment initiation request for an online payment. Returns an object with a URL to display an online payment page to the shopper.
+
+#### PayRequest - Payment Request
+```java
+public class PayRequest {
+    private String amount = null;
+    private Boolean showTips = null;
+    private PaymentMethod paymentMethod = null;
+}
+
+public class PaymentMethod {
+  private PaymentMethodType paymentMethodType = null;
+}
+
+public enum PaymentMethodType {
+  CARD("CARD"),
+  ALIPAY("ALIPAY"),
+  WECHAT("WECHAT")
+}
 ```
 
-## Samples
+Request parameters:
+- `store_id` - the ID of your assigned store
+- `payment_reference` - a unique reference for the payment, sent from the client and returned by the server
+- `amount` - the amount to request for the online payment, formatted as a 2.dp decimal number, such as `1.00`
+- `currency` - the currency for this payment in 3 character ISO format, such as `GBP`
+- `order_id` - a unique order ID for this payment, sent from the client and returned by the server
+- `return_url` - where the payment form will redirect to after the payment has completed, the return url will have additional query parameters appended to indicate the status of the payment request.
+- `showTimer` - flag to show countdown timer in payment page
+- `expiry` - how long the payment form will wait until the payment expires and the page will redirect to the return url
+
+
+#### PayResponse : Payment Response
+
+````java
+public enum ErrorType {
+  UNKNOWN("UNKNOWN"),
+  DUPLICATE_ATTEMPT("DUPLICATE_ATTEMPT"),
+  INVALID_REQUEST("INVALID_REQUEST"),
+  CANCELLED("CANCELLED")
+}
+
+public class Error {
+  private ErrorType type;       //Enum for the error type
+  private String message;       //Error message
+}
+
+public class PaymentInitiationResponse {
+    private Response response;
+    private Error error;
+}
+
+public class Response {
+  private String paymentId;     // Unique identifier created by Kody
+  private String paymentUrl;    // The URL to send to the user from your application 
+}
+````
+
+#### Java Demo
+````java
+import common.PaymentClient;
+import com.kodypay.grpc.pay.v1.*;
+
+PaymentClient paymentClient = new PaymentClient(HOSTNAME, APIKEY);
+
+String storeId = "UUID of assigned store";
+String orderId = "UUID of assigned store";
```

Comment:
Order ID - may be shared for multiple payments (please check whether that is correct)
Assuming that `paymentReference` should the unique ID